toolkit/src/main/cpp/editor/meta_export_template_dialog.cpp

```diff
@@ -0,0 +1,323 @@
+				r_status += " " + String::humanize_size(p_request->get_downloaded_bytes()) + "/" + String::humanize_size(p_request->get_body_size());
+			} else {
+				r_status += " " + String::humanize_size(p_request->get_downloaded_bytes());
+			}
+			break;
+		case HTTPClient::STATUS_CONNECTION_ERROR:
+			r_status = "Connection Error";
+			success = false;
+			break;
+		case HTTPClient::STATUS_TLS_HANDSHAKE_ERROR:
+			r_status = "TLS Handshake Error";
+			success = false;
+			break;
+	}
+
+	return success;
+}
+
+void MetaExportTemplateDialog::show() {
+	popup_centered();
+
+	download_progress_bar->hide();
+	message->add_theme_color_override("font_color", get_theme_color(StringName("font_color"), StringName("Label")));
+
+	if (DirAccess::dir_exists_absolute(META_EXPORT_TEMPLATE_PREBUILT_DIR)) {
+		download_button->set_disabled(true);
+		message->set_text(META_EXPORT_TEMPLATE_ALREADY_INSTALLED);
+	} else {
+		download_button->set_disabled(false);
+		message->set_text(META_EXPORT_TEMPLATE_MISSING);
+	}
+}
+
+void MetaExportTemplateDialog::_download_template() {
+	if (is_downloading_template) {
+		return;
+	}
+	is_downloading_template = true;
+
+	if (download_request) {
+		memfree(download_request);
+	}
+
+	download_request = memnew(HTTPRequest);
+	add_child(download_request);
+	download_request->connect("request_completed", callable_mp(this, &MetaExportTemplateDialog::_download_request_completed));
+
+	download_progress_bar->show();
+	download_progress_bar->set_indeterminate(true);
+	_set_current_progress_status("Starting the download...");
+
+	download_request->set_download_file(EditorInterface::get_singleton()->get_editor_paths()->get_cache_dir().path_join("tmp-meta-export-template.zip"));
+	download_request->set_use_threads(true);
+
+	EditorInterface *editor_interface = EditorInterface::get_singleton();
+	if (editor_interface == nullptr) {
+		_set_current_progress_status(vformat("Failed to get editor interface."), true);
+		return;
+	}
+
+	Ref<EditorSettings> editor_settings = EditorInterface::get_singleton()->get_editor_settings();
+	if (editor_settings.is_null() || !editor_settings->has_setting("network/http_proxy/host") || !editor_settings->has_setting("network/http_proxy/port")) {
+		_set_current_progress_status(vformat("Failed to get editor settings."), true);
+		return;
+	}
+
+	const String proxy_host = editor_settings->get_setting("network/http_proxy/host");
+	const int proxy_port = editor_settings->get_setting("network/http_proxy/port");
+	download_request->set_http_proxy(proxy_host, proxy_port);
+	download_request->set_https_proxy(proxy_host, proxy_port);
+
+	if (!editor_settings->has_setting("xr/meta_toolkit/base_download_url")) {
+		_set_current_progress_status(vformat("No base download URL for Meta toolkit found in editor settings."), true);
+		return;
+	}
+
+	String download_url = (String)editor_settings->get_setting("xr/meta_toolkit/base_download_url") + GODOT_META_TOOLKIT_VERSION + "/meta-export-template.zip";
```

Comment:
The Godot version should be included in the filename, so that we match the editor version with the template version. So, something like:

```suggestion
	const GDExtensionGodotVersion2 &godot_version = godot::internal::godot_version;
	String filename = vformat("meta-export-template-v%d.%d.%d-%s", godot_version.major, godot_version.minor, godot_version.patch, godot_version.status);
	String download_url = (String)editor_settings->get_setting("xr/meta_toolkit/base_download_url") + GODOT_META_TOOLKIT_VERSION + "/" + filename;
```

We may need to update godot-cpp to get `GDExtensionGodotVersion2` which has the `.status` field

Reply:
I've updated the `godot-cpp` version and applied this change!